packages/discord.js-utilities/src/lib/type-guards.ts

```diff
@@ -82,3 +94,29 @@ export function isVoiceChannel(channel: Channel): channel is VoiceChannel {
 export function isStageChannel(channel: Channel): channel is StageChannel {
 	return channel.type === 'GUILD_STAGE_VOICE';
 }
+
+/**
+ * Checks whether a given channel is a {@link ThreadChannel}
+ * @param channel The channel to check.
+ */
+export function isThreadChannel(channel: Channel): channel is ThreadChannel {
+	return channel.isThread();
+}
+
+/**
+ * Checks whether a given channel allows NSFW content or not
+ * @param channel The channel to check whether it allows NSFW or not
```

Comment:
```suggestion
 * @param channel The channel to check.
```
For consistency and conciseness